mono/mini/mini-amd64.c

```diff
@@ -7562,7 +7562,7 @@ mono_arch_emit_exceptions (MonoCompile *cfg)
 #endif /* DISABLE_JIT */
 
 void*
-mono_arch_instrument_prolog (MonoCompile *cfg, void *func, void *p, gboolean enable_arguments)
+(mono_arch_instrument_prolog) (MonoCompile *cfg, void *func, void *p, gboolean enable_arguments)
```

Comment:
Stretching the definition of "allocator", perhaps, but it a void*-returning function that users assign result to non-void* (guint8*). We have a mix of void* and guint8*.

Reply:
This function is also defined in `mini-sparc.c`, `mini-s390x.c`, `mini-mips.c`, etc.

My preference would be not to fixup `mono_arch_instrument_prolog`/epilog in this PR.  Let's just get `g_cast_t<>` merged in the cases where we need it.